src/services/getEditsForFileRename.ts

```diff
@@ -1,65 +1,180 @@
 /* @internal */
 namespace ts {
-    export function getEditsForFileRename(program: Program, oldFilePath: string, newFilePath: string, host: LanguageServiceHost, formatContext: formatting.FormatContext): ReadonlyArray<FileTextChanges> {
-        const pathUpdater = getPathUpdater(oldFilePath, newFilePath, host);
+    export function getEditsForFileRename(program: Program, oldFileOrDirPath: string, newFileOrDirPath: string, host: LanguageServiceHost, formatContext: formatting.FormatContext, preferences: UserPreferences): ReadonlyArray<FileTextChanges> {
+        const useCaseSensitiveFileNames = hostUsesCaseSensitiveFileNames(host);
+        const getCanonicalFileName = createGetCanonicalFileName(useCaseSensitiveFileNames);
+        const oldToNew = getPathUpdater(oldFileOrDirPath, newFileOrDirPath, getCanonicalFileName);
+        const newToOld = getPathUpdater(newFileOrDirPath, oldFileOrDirPath, getCanonicalFileName);
         return textChanges.ChangeTracker.with({ host, formatContext }, changeTracker => {
-            updateTsconfigFiles(program, changeTracker, oldFilePath, newFilePath);
-            for (const { sourceFile, toUpdate } of getImportsToUpdate(program, oldFilePath, host)) {
-                const newPath = pathUpdater(isRef(toUpdate) ? toUpdate.fileName : toUpdate.text);
-                if (newPath !== undefined) {
-                    const range = isRef(toUpdate) ? toUpdate : createStringRange(toUpdate, sourceFile);
-                    changeTracker.replaceRangeWithText(sourceFile, range, isRef(toUpdate) ? newPath : removeFileExtension(newPath));
-                }
-            }
+            updateTsconfigFiles(program, changeTracker, oldToNew, newFileOrDirPath, host.getCurrentDirectory(), useCaseSensitiveFileNames);
+            updateImports(program, changeTracker, oldToNew, newToOld, host, getCanonicalFileName, preferences);
         });
     }
 
-    function updateTsconfigFiles(program: Program, changeTracker: textChanges.ChangeTracker, oldFilePath: string, newFilePath: string): void {
-        const configFile = program.getCompilerOptions().configFile;
+    /** If 'path' refers to an old directory, returns path in the new directory. */
+    type PathUpdater = (path: string) => string | undefined;
+    function getPathUpdater(oldFileOrDirPath: string, newFileOrDirPath: string, getCanonicalFileName: GetCanonicalFileName): PathUpdater {
+        const canonicalOldPath = getCanonicalFileName(oldFileOrDirPath);
+        return path => {
+            const canonicalPath = getCanonicalFileName(path);

```

Comment:
This still handles the path as is but you need to be able to handle include/exclude/file like: 
"files": "src/foo.ts" or "./src/foo.ts",  "../src/foo.ts" etc?

Reply:
For include/exclude we don't just call this directly, see `tryUpdateString`. First we prepend the config directory to the path, call oldToNew, then we take it out again after.